### PR TITLE
Get Workspace by ID

### LINKED ISSUE #2529: GET workspace/<workspace_id> fails on auto_create workspace when no scope_id is set
When creating a new workspace (via the UI, or Swagger) and setting the `client_id` to `auto_create`, any subsequent calls to `workspaces/new_workspace_id` fail whilst the workspace is being created. 

Relevant call stack from API logs: 
```
File "/api/./api/routes/workspaces.py", line 79, in retrieve_workspace_by_workspace_id
2022-08-30T13:46:16.623829724Z if access_service.get_workspace_role(user, workspace, user_role_assignments) != WorkspaceRole.NoRole:
2022-08-30T13:46:16.623833324Z File "/api/./services/aad_authentication.py", line 353, in get_workspace_role
2022-08-30T13:46:16.623836924Z raise AuthConfigValidationError(strings.AUTH_CONFIGURATION_NOT_AVAILABLE_FOR_WORKSPACE)
2022-08-30T13:46:16.623840624Z services.access_service.AuthConfigValidationError: Auth configuration not available for workspace
```

Be good to get any quick thoughts from @stuartleeks / @jjgriff93 as you both made some auth-related changes to the `retrieve_workspace_by_workspace_id` method. I'm happy to pick the bug up but any pointers appreciated.

### PR DESCRIPTION
**Pull request by damoodamoo**
# Resolves #2529 

## What is being addressed
- When a TRE admin creates a new workspace using `auto_create`, there is no AAD app to authenticate against, but we still need to return the workspace details for display in the UI. Now TRE Admins are allowed to fetch any workspace by ID, using a core token.
- When a researcher user lands on a workspace page, they need to be able to fetch the workspace without knowing the `scope_id`. They couldn't do this as they weren't a TRE Admin and they didn't have a workspace token. Since the `scope_id` is consistently named, in the UI we can infer the value and authenticate the user against it to get a workspace token to allow them to then fetch the details.

### CODE REVIEW

File: api_app/tests_ma/test_api/test_routes/test_workspaces.py [hunk 1]
```diff
@@ -272,32 +270,14 @@ async def test_get_workspace_by_id_get_returns_404_if_resource_is_not_found(self
...
-    # [GET] /workspaces/{workspace_id}
-    @patch("api.dependencies.workspaces.WorkspaceRepository.get_workspace_by_id")
-    @patch("api.routes.workspaces.get_identity_role_assignments")
-    async def test_get_workspace_by_id_get_returns_422_if_workspace_id_is_not_a_uuid(self, access_service_mock, _, app, client):
-        access_service_mock.return_value = [RoleAssignment('ab123', 'ab124')]
-        response = await client.get(app.url_path_for(strings.API_GET_WORKSPACE_BY_ID, workspace_id="not_valid"))
-        assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
-
-    @patch("api.dependencies.workspaces.WorkspaceRepository.get_workspace_by_id")
-    @patch("api.routes.workspaces.ResourceTemplateRepository.get_templates_information", return_value=[ResourceTemplateInformation(name="test")])
```
Comment (tanya-borisova) [inline api_app/tests_ma/test_api/test_routes/test_workspaces.py:284]: Think this and the following tests are now removed (looking at locally), could you add them back

**Review by tanya-borisova — CHANGES_REQUESTED** [review]

File: api_app/tests_ma/test_api/test_routes/test_workspaces.py (hunk 1 above)
Comment (damoodamoo) [reply api_app/tests_ma/test_api/test_routes/test_workspaces.py:284]: cheers:
- the `422_` test has been moved elsewhere as requires a workspace user 
- the workspace service + user resource template GET tests have been moved into the ResearcherUser class as they now require a workspace user

**Review by tanya-borisova — APPROVED** [review]

**Comment by damoodamoo** [comment]
/test-force-approve
have previously run and passed extended tests

**Pull request merged**